platform/core/src/services/ToolBarService/ToolBarService.js

```diff
@@ -154,12 +154,15 @@ export default class ToolBarService {
       if (btn.props.clickHandler) {
         btn.clickHandler(evt, btn, btnSection);
       }
+      if (props && props.onClick) {
+        props.onClick(evt, btn, btnSection, props);
+      }
     };
 
     return {
       id,
       Component: component || buttonType.defaultComponent,
-      componentProps: Object.assign({}, props, { onClick }), //
+      componentProps: Object.assign({}, btn.props, props, { onClick }), //
```

Comment:
This is tricky. I wouldn't just override the `onClick` you intend to set in the `viewer` project for all buttons. It might make more sense to wrap the buttons `onClick`, if it has one? Kind of chaining them, so you don't squash some user defined functionality by accident.

Each toolbar module button definition can optionally provide a `clickHandler`. We have some examples of these in: `extensions/default/src/getToolbarModule`

When our UI binds `onClick` for a button component (here), it does:

buttonType.clickHandler -->
btn.props.onClick -->
btn.props.clickHandler

When you call `getButtonSection`, you're passing in `props: { onClick: fn }` and adding your onClick to the bottom of that chain of callbacks (from 3x to 4x).

buttonType.clickHandler -->
btn.props.onClick -->
btn.props.clickHandler -->
props.onClick (from `viewer`)

------

Just thinking out loud.